File: 2208/solution.py
from heapq import heapify, heappop, heappush
from typing import List


class Solution:
    def halveArray(self, nums: List[int]) -> int:
        n, total = len(nums), 0
        for i in range(n):
            total += nums[i]
            nums[i] *= -1
        count = 0
        res = 0
        heapify(nums)
        while count < total / 2:
            num = heappop(nums)
            num /= 2
            count -= num
            res += 1
            heappush(nums, num)
        return res

File: 2208/Solution.java
import java.util.PriorityQueue;

class Solution {
    public int halveArray(int[] nums) {
        PriorityQueue<Double> queue = new PriorityQueue<Double>((a, b) -> {
            return b.compareTo(a);
        });
        double total = 0;
        int res = 0;
        for (int num : nums) {
            queue.add(Double.valueOf(num));
            total += num;
        }
        double count = 0;
        while (count < total / 2.0) {
            double num = queue.poll();
            num /= 2;
            count += num;
            res++;
            queue.add(num);
        
        }
        return res;
    }
}